sdk/eventgrid/azure-eventgrid/azure/eventgrid/_consumer.py

```diff
@@ -38,7 +39,9 @@ def decode_cloud_event(self, cloud_event, **kwargs):
         encode = kwargs.pop('encoding', 'utf-8')
         try:
             cloud_event = CloudEvent._from_json(cloud_event, encode)
-            deserialized_event = CloudEvent.deserialize(cloud_event)
+            deserialized_event = CloudEvent._from_generated(
+                InternalCloudEvent.deserialize(cloud_event)
```

Comment:
It's a weird feeling for me that the caller has to know both about the CloudEvent and the InternalCloudEvent.  Would it not offer better separation of concerns/isolation to have the InternalCloudEvent.deserialize within the _from_generated so the caller never sees anything but our "nice and pretty" object contract?